app/components/PredictionMarkets/PredictionMarkets.jsx

```diff
@@ -0,0 +1,388 @@
+import ResolveModal from "./ResolveModal";
+import {ChainStore} from "bitsharesjs";
+import {Button} from "bitshares-ui-style-guide";
+
+export default class PredictionMarkets extends Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            assets: [],
+            lastAssetSymbol: null,
+            markets: [],
+            currentAccountId: null,
+            searchTerm: "",
+            detailsSearchTerm: "",
+            selectedMarket: null,
+            opinions: [],
+            preselectedOpinion: "no",
+            preselectedAmount: 0,
+            isCreateMarketModalOpen: false,
+            isAddOpinionModalOpen: false,
+            isResolveModalOpen: false
+        };
+        this._updateAssetsList("A");
+    }
+
+    componentWillReceiveProps(np) {
+        if (np.currentAccount !== this.props.currentAccount) {
+            this.setState({
+                currentAccountId: ChainStore.getAccount(np.currentAccount).get(
+                    "id"
+                )
+            });
+        }
+        let searchAsset = this.state.lastAssetSymbol;
+        if (np.assets) {
+            const lastAsset = np.assets
+                .sort((a, b) => {
+                    if (a.symbol > b.symbol) {
+                        return 1;
+                    } else if (a.symbol < b.symbol) {
+                        return -1;
+                    } else {
+                        return 0;
+                    }
+                })
+                .last();
+            searchAsset = lastAsset ? lastAsset.symbol : "A";
+            const assets = np.assets.filter(
+                a => a.bitasset_data && a.bitasset_data.is_prediction_market
+            );
+            this.setState({
+                assets: [...assets]
+            });
+            this.updateMarketsList();
+        }
+        if (
+            !this.state.lastAssetSymbol ||
+            this.state.lastAssetSymbol !== searchAsset
+        ) {
+            this._updateAssetsList(searchAsset);
+            this.setState({
+                lastAssetSymbol: searchAsset
+            });
+        }
+        console.log(this.props.markets);
+        const asks = this.props.markets.asks.map(element => ({
+            order_id: element.id,
+            opinionator: element.seller,
+            opinion: "yes",
+            amount: element.for_sale,
+            fee: element.fee
+        }));
+        const bids = this.props.markets.bids.map(element => ({
+            order_id: element.id,
+            opinionator: element.seller,
+            opinion: "no",
+            amount: element.for_sale,
+            fee: element.fee
+        }));
+        this.setState({opinions: [...asks, ...bids]});
+    }
+
+    async _updateAssetsList(lastAsset) {
+        AssetActions.getAssetList.defer(lastAsset, 100);
+    }
+
+    updateMarketsList() {
+        const markets = this.state.assets.map(item => ({
```

Comment:
did not check if that is done, but the markets can be restricted to: PREDICTIONASSET / ITSCOLLATERAL. For exmaple: If SOMEPREDICTION is using BTS as collateral, only that market needs to be shown.

Keep the list based approach nonetheless for future expansion

Reply:
Could you describe in more detail, it is not entirely clear, what restriction do you mean?

Reply:
It's only necessary to subscripte to the PREDICTIONASSET / ITSCOLLATERAL market.

Reply:
In this function updateMarketsList() we form the list of prediction markets from list of assets to be shown in UI.
Elsewhere in the code we take all the assets and filter them checking the flag "bitasset_data.is_prediction_market".
Do we need any other filters for prediction markets?
As far as we understand, in expression PREDICTIONASSET / ITSCOLLATERAL predictionasset is the symbol of prediction market, and itscollateral is its backing asset.

Reply:
Yes, that is correct. My comment refers to making sure that we `ONLY` subscribe to the market of the currently selected prediction market asset, and not all.

Reply:
The code has the following logic: we get all the assets from the Chain, filter the list to get only the prediction markets and show the list in the Prediction Markets table. After clicking on one of the rows, we subscribe to the selected market and get the list of limit orders, then we show it in the Opinions table. So seems it's done.